server/internal/heavy/components.go

```diff
@@ -234,6 +252,10 @@ func newComponents(ctx context.Context, cfg configuration.Configuration, genesis
 		drops := drop.NewDB(DB)
 		jets := jet.NewDBStore(DB)
 		jetKeeper := replica.NewJetKeeper(jets, DB)
+		err := truncateNotFinalizedData(ctx, jetKeeper, drops)
```

Comment:
Components are for initializing stuff, not performing work. We should not truncate it here. Let's find an appropriate component.